dnSpy/dnSpy/Documents/TreeView/DocumentTreeViewSettings.cs

```diff
@@ -100,6 +100,18 @@ class DocumentTreeViewSettings : ViewModelBase, IDocumentTreeViewSettings {
 		}
 		bool deserializeResources = true;
 
+		public DocumentFilterType FilterDraggedItems {
+			get { return filterDraggedItems; }
+			set {
+				if (filterDraggedItems != value) {
+					filterDraggedItems = value;
+					OnPropertyChanged(nameof(FilterDraggedItems));
+					OnModified();
+				}
+			}
+		}
+		DocumentFilterType filterDraggedItems = DocumentFilterType.DotNetAndPE;
```

Comment:
There should be a new value "any supported file". It's possible for an extension to support reading eg. XML files or some other type of file. "any supported file" should be default. This will normally only allow .NET and PE files.